### DISCUSSION TITLE
re_replace - Found unknown escape character '.'

### DISCUSSION THREAD

**Discussion by ghost** [Flexget/Flexget#3680] [Q&A] [answered]
I'm trying to get rid of instances of file formats repeated into filenames by mistake (ex: `Movie_123.mkv.mkv`) using `set` but I'm not able to use `\.` in my pattern. The console will return the error specified in the title. How can I use proper regex patterns in YAML with `re_replace`?

```yml
set:
  content_filename: "{{ title|re_replace('[\\/:*?\"<>|]', '')|re_replace('\.mkv$', '') }}"
```

**Comment by BrutuZ** [comment, marked as answer]
You may have to escape `\` up to three times (`\\\\`).
I can never tell how many are enough for each use-case though, other than having a few `\\\\b` on my config for word boundaries I usually start with a couple since sometimes that's enough and add more as I test it 😅

**Reply by gazpachoking** [reply]
Yeah, the issue is that there are multiple possible levels interpreting the backslashes as escape sequences. YAML, jinja and regex. You can avoid yaml interpreting backslashes as escapes by using single quotes rather than double, which at least gives one less level of backslashes needed.

**Reply by ghost** [reply]
Escaping a single time while wrapped in single quotes did the trick. Thanks!
```yml
re_replace('\\.mkv$', '')
```
```
2023-02-10 17:36:33 VERBOSE  check                         Config passed check.
```

**Reply by BrutuZ** [reply]
Make sure it is matching what you want besides passing the check.
It might be looking for the literal slash before any character + mkv in the string instead of a literal dot\period + mkv

**Reply by ghost** [reply]
Yep. Just found out the hard way. I guess I'll keep adding `\` till it starts working
